pwiz_tools/Skyline/Model/Results/SpectrumFilter.cs

```diff
@@ -101,6 +103,8 @@ public sealed class SpectrumFilter : IFilterInstrumentInfo
                 _isWatersFile = instrumentInfo.IsWatersFile;
                 _isWatersSonar = instrumentInfo.IsWatersSonarData;
                 _configInfoList = instrumentInfo.ConfigInfoList;
+                _fileContentList = instrumentInfo.FileContentList;
+                _sourceHasDeclaredMS2Scans = _fileContentList.Contains(@"MSn spectrum");
```

Comment:
I recommend going with CVID so if the term name ever changes in the future we won't need to support multiple strings (the old string for existing mzMLs and the new string for vendor files and new mzMLs).

Reply:
Makes sense. I was following a pattern I'd seen elsewhere but I think you're right.

Reply:
Whoops I didn't notice this file was outside MsDataFileImpl. Hmm. Maybe a dedicated property for HasDeclaredMSn?

Reply:
Yeah, probably a more robust and succinct approach.

Reply:
Nevermind. It's very unlikely this particular term will change name, and we could always switch to a property later if it does.

Reply:
Oops, too late. I like your way better, anyway.